src/ldp/alg/rollout.py

```diff
@@ -101,6 +102,7 @@ def __init__(
 
         self.traj_buffer: dict[str, Trajectory] = {}
         self.callbacks = callbacks or []
```

Comment:
Why can't a normal `Callback` just edit the message? They are mutable

It's not clear to me that message mutation should only take place in `after_transition`.

I guess, it seems we don't need this special case, we can already accomplish the use case with normal `Callback`.

If what you want is ordering, maybe we should instead add a priority concept to `Callback`. Feels more general